cmd/create/oidcprovider/cmd.go

```diff
@@ -0,0 +1,261 @@
+	rprtr "github.com/openshift/rosa/pkg/reporter"
+)
+
+var modes []string = []string{"auto", "manual"}
+
+var args struct {
+	clusterKey string
+	mode       string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "oidc-provider",
+	Aliases: []string{"oidcprovider"},
+	Short:   "Create OIDC provider for an STS cluster.",
+	Long:    "Create OIDC provider for operators to authenticate against an STS cluster.",
+	Example: `  # Create OIDC provider for cluster named "mycluster"
+  rosa create oidc-provider --cluster=mycluster`,
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVarP(
+		&args.clusterKey,
+		"cluster",
+		"c",
+		"",
+		"Name or ID of the cluster to create the roles for (required).",
+	)
+	Cmd.MarkFlagRequired("cluster")
+
+	flags.StringVar(
+		&args.mode,
+		"mode",
+		modes[0],
+		"How to perform the operation. Valid options are:\n"+
+			"auto: Roles will be created using the current AWS account\n"+
+			"manual: Role files will be saved in the current directory",
+	)
+	Cmd.RegisterFlagCompletionFunc("mode", modeCompletion)
+
+	confirm.AddFlag(flags)
+	interactive.AddFlag(flags)
+}
+
+func modeCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return modes, cobra.ShellCompDirectiveDefault
+}
+
+func run(cmd *cobra.Command, _ []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Check that the cluster key (name, identifier or external identifier) given by the user
+	// is reasonably safe so that there is no risk of SQL injection:
+	clusterKey := args.clusterKey
+	if !ocm.IsValidClusterKey(clusterKey) {
+		reporter.Errorf(
+			"Cluster name, identifier or external identifier '%s' isn't valid: it "+
+				"must contain only letters, digits, dashes and underscores",
+			clusterKey,
+		)
+		os.Exit(1)
+	}
+
+	// Determine if interactive mode is needed
+	if !interactive.Enabled() && !cmd.Flags().Changed("mode") {
+		interactive.Enable()
+	}
+
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+
+	creator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get IAM credentials: %s", err)
+		os.Exit(1)
+	}
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	// Try to find the cluster:
+	reporter.Debugf("Loading cluster '%s'", clusterKey)
+	cluster, err := ocmClient.GetCluster(clusterKey, creator)
+	if err != nil {
+		reporter.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	if cluster.State() != cmv1.ClusterStatePending {
```

Comment:
Is the only reason for a cluster to be pending OIDC configuration? Can we be more specific and look at the `oidc_ready` status? Should we also validate that this cluster is an STS cluster?

Reply:
It doesn't really matter. Currently 'Pending' means waiting for Account Claims or OIDC, however the OIDC can be done at any stage _prior_ to 'Installing' (which implies OIDC is already set up). If the OIDC is set up before it gets to the 'Waiting for OIDC configuration' state, then it just moves on to 'Installing' right away.

Think of it as a `if cluster.State() < cmv1.ClusterStateInstalling`, except for the cluster states are not numeric

Reply:
> Should we also validate that this cluster is an STS cluster?
We already do that in the next condition (`cluster.AWS().STS().RoleARN() == ""`)